hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3A.java

```diff
@@ -37,7 +37,8 @@ public class S3A extends DelegateToFileSystem {
 
   public S3A(URI theUri, Configuration conf)
       throws IOException, URISyntaxException {
-    super(theUri, new S3AFileSystem(), conf, "s3a", false);
+    super(theUri, new S3AFileSystem(), conf,
+        theUri.getScheme().isEmpty() ? "s3a" : theUri.getScheme(), false);
```

Comment:
move this to a constant in `org.apache.hadoop.fs.s3a.impl.InternalConstants` and then you can reference it in tests.